crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs

```diff
@@ -108,10 +110,19 @@ pub(crate) fn mutable_argument_default(checker: &mut Checker, function_def: &ast
             .map(|target| QualifiedName::from_dotted_name(target))
             .collect();
 
-        if is_mutable_expr(default, checker.semantic())
+        // Either the expression _or_ the annotation must be mutable.
+        if (is_mutable_expr(default, checker.semantic())
             && !parameter.annotation.as_ref().is_some_and(|expr| {
                 is_immutable_annotation(expr, checker.semantic(), extend_immutable_calls.as_slice())
-            })
+            }))
+            || (parameter.annotation.as_ref().is_some_and(|annotation| {
+                is_mutable_annotation(annotation, checker.semantic())
+                    && !is_immutable_expr(
+                        default,
+                        checker.semantic(),
+                        extend_immutable_calls.as_slice(),
+                    )
+            }))
```

Comment:
(At the very least I think these should return enums rather than bools, with a single call to check immutability.)